frontend-new/components/TransferModal.tsx

```diff
@@ -0,0 +1,51 @@
+import { Typography } from '@mui/material'
+import Button from '@mui/material/Button'
+import TextField from '@mui/material/TextField'
+import { FC } from 'react'
+
+import { useGetTransferToken, useClipboard, useFullShortPath } from 'app/hooks'
+import { media } from 'app/styles/theme'
+import { Link } from 'app/types'
+
+import { Modal } from './BaseModal'
+
+interface Props {
+  open: boolean
+  onClose: () => void
+  link: Link
+}
+
+export const TransferModal: FC<Props> = ({ open, onClose, link }) => {
+  const { namespace, shortpath, id } = link
+  const transferToken = useGetTransferToken(id)
+  const handleCopy = useClipboard(transferToken?.url || '')
```

Comment:
If transferToken is not valid how do we handle this error?   Currently seems like the user would be copying an empty string?

Reply:
disabled a button as it is no sense to copy empty string